### PR TITLE
feat (tf/pt): add atomic weights to tensor loss

### PR DESCRIPTION
**Pull request by ChiahsinChu**
Interfaces are of particular interest in many studies. However, the configurations in the training set to represent the interface normally also include large parts of the bulk material. As a result, the final model would prefer the bulk information while the interfacial information is less learnt. It is difficult to simply improve the proportion of interfaces in the configurations since the electronic structures of the interface might only be reasonable with a certain thickness of bulk materials. Therefore, I wonder whether it is possible to define weights for atomic quantities in loss functions. This allows us to add higher weights for the atomic information for the regions of interest and probably makes the model "more focused" on the region of interest. 
In this PR, I add the keyword `enable_atomic_weight` to the loss function of the tensor model. In principle, it could be generalised to any atomic quantity, e.g., atomic forces. 
I would like to know the developers' comments/suggestions about this feature. I can add support for other loss functions and finish unit tests once we agree on this feature. 

Best. 

## Summary by CodeRabbit

- **New Features**
	- Introduced an optional parameter for atomic weights in loss calculations, enhancing flexibility in the `TensorLoss` class.
	- Added a suite of unit tests for the `TensorLoss` functionality, ensuring consistency between TensorFlow and PyTorch implementations.

- **Bug Fixes**
	- Updated logic for local loss calculations to ensure correct application of atomic weights based on user input.

- **Documentation**
	- Improved clarity of documentation for several function arguments, including the addition of a new argument related to atomic weights.

### CODE REVIEW

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
Please provide the unit test for the feature. 

do you consider implementing it to the pt backend?

**Comment by ChiahsinChu** [comment]
> Please provide the unit test for the feature.
> 
> do you consider implementing it to the pt backend?

In this PR, this feature is implemented for both tf and pt backend.

**Comment by njzjz** [comment]
I am not sure if we need to wait for #4105.

**Review by njzjz — APPROVED** [review]
I approve the current PR, but I am not sure when #4105 can be ready.

**Comment by wanghan-iapcm** [comment]
> I approve the current PR, but I am not sure when #4105 can be ready.

I think #4105 should not block the merge of this pr.

**Review by wanghan-iapcm — APPROVED** [review]
